servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletOutputStream.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.servlet;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.WriteListener;
+
+import com.linecorp.armeria.common.HttpData;
+
+/**
+ * Servlets output streams (wrapper classes) that control access to the flow.
+ */
+final class DefaultServletOutputStream extends ServletOutputStream {
+    private final DefaultServletHttpResponse response;
+
+    DefaultServletOutputStream(DefaultServletHttpResponse response) {
+        requireNonNull(response, "response");
+        this.response = response;
+    }
+
+    @Override
+    public boolean isReady() {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public void setWriteListener(WriteListener listener) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public void close() throws IOException {
+    }
+
+    @Override
+    public void flush() throws IOException {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public void write(int b) throws IOException {
+        final byte[] bytes = new byte[1];
+        bytes[0] = (byte) b;
+        write(bytes);
+    }
+
+    @Override
+    public void write(byte[] b, int off, int len) throws IOException {
+        requireNonNull(b, "b");
+        checkArgument(off >= 0, "off: %s (expected: >= 0)", off);
+        checkArgument(len >= 0, "len: %s (expected: >= 0)", len);
+        response.write(HttpData.copyOf(Arrays.copyOfRange(b, off, len)));
```

Comment:
What happens to the second right? A user can do it?

Reply:
Do you mean `off >= 0` is right?

Reply:
Let's say that if a user calls:
```java
DefaultServletOutputStream stream = ...
stream.write(1);
stream.write(2);
```
The first `stream.write(1);` call will close the response stream writer by calling `response.getResponseWriter().close();`
So 2 cannot be written. Isn't it?

Reply:
Yes, If want to write multiple part of download file , How to write multiple time? Can I call tryWrite() multiple time?

Reply:
If we can't write multiple time, I think write(2) should be raise an exception. Ofcourse, we need a `AtomicBoolean isWritten` to check response have already written. What do you think?

Reply:
I think we should allow users call write multiple time.
How about caching the written bytes and write it to the `HttpResponseWriter` when `close()` or `flush()` is called?